### ISSUE TITLE
SchemaValidator limits property keys too much?

### ISSUE THREAD

**Issue by steven-aerts** [RepreZen/KaiZen-OpenApi-Parser#96]
The schema validator currently [limits](https://github.com/RepreZen/KaiZen-OpenApi-Parser/blob/f408e14486d75f8515a9d79b1fe4e0c68a8dec79/kaizen-openapi-parser/src/main/java/com/reprezen/kaizen/oasparser/val3/SchemaValidator.java#L59) the possible key values of a map to *NOEXT_NAME_REGEX* which means `(?!x-)[a-zA-Z0-9\\._-]+`.

I think this is too strict, I think it should allow nearly all characters.

If you agree, I will happily provide a pull request.

**Comment by andylowry** [comment]
@steven-aerts The regex is based on the [OpenAPI 3.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#componentsObject), where most of the maps have keys that are restricted to a regular expression. Here's the relevant excerpt:

```
All the fixed fields declared above are objects that MUST use keys that match the regular expression: ^[a-zA-Z0-9\.\-_]+$.
```

The exclusion of names starting with `x-` is used where the map value also allows for vendor extensions, which always start with `x-`.

If you think this restriction has been applied where it shouldn't be, let me know.

**Comment by andylowry** [comment]
Clarification: Issue specifically involved the `properties` map in a `Schema Object`, where no restriction on key values is evident in OAS3.0.

This issue is addressed in #97, contributed by @steven-aerts.

**Comment by steven-aerts** [comment]
Thanks Andy
